roles/pulp_django_user/tasks/user_groups/add_or_remove_users.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+- name: Get all groups that user belongs to
+  uri:
+    url: "{{ pulp_get_user_url }}?username={{ user.username }}"
+    user: "{{ pulp_admin_username }}"
+    password: "{{ pulp_admin_password }}"
+    method: GET
+    status_code: 200
+    force_basic_auth: true
+  register: user_group_result
+
+- name: Set current_group_membership fact
+  set_fact:
+    current_group_membership: "{{ user_group_result | json_query('json.results[*].groups') | flatten | map(attribute='name') | list }}"
+
+- name: Remove user from groups not defined in pulp_django_users
+  include_tasks: remove_user_from_groups.yml
+  loop: "{{ current_group_membership | difference(user.groups | default([])) }}"
```

Comment:
nit: I think you need `default([], true)` to make this work:

> Adding an empty `groups:` for a user will result in that user being removed from all groups.

Reply:
You're right - fixed in https://github.com/stackhpc/ansible-collection-pulp/pull/31/commits/2794ce64d33d9c63fddc38636d9a89dd0cfba397